x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/persistence/ElasticsearchMappings.java

```diff
@@ -95,6 +110,278 @@
+
+        .startObject(Job.CUSTOM_SETTINGS.getPreferredName())
+            // Custom settings are an untyped map
+            .field(ENABLED, false)
+        .endObject()
+
+        .startObject(Job.DATA_DESCRIPTION.getPreferredName())
+            .startObject(PROPERTIES)
+                .startObject(DataDescription.FORMAT_FIELD.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+                .startObject(DataDescription.TIME_FIELD_NAME_FIELD.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+                .startObject(DataDescription.TIME_FORMAT_FIELD.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+                .startObject(DataDescription.FIELD_DELIMITER_FIELD.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+                .startObject(DataDescription.QUOTE_CHARACTER_FIELD.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+            .endObject()
+        .endObject()
+
+        .startObject(Job.DESCRIPTION.getPreferredName())
+            .field(TYPE, TEXT)
+        .endObject()
+        .startObject(Job.FINISHED_TIME.getPreferredName())
+            .field(TYPE, DATE)
+        .endObject()
+        .startObject(Job.LAST_DATA_TIME.getPreferredName())
+            .field(TYPE, DATE)
+        .endObject()
+        .startObject(Job.ESTABLISHED_MODEL_MEMORY.getPreferredName())
+            .field(TYPE, LONG) // TODO should be ByteSizeValue
+        .endObject()
+
+        .startObject(Job.MODEL_PLOT_CONFIG.getPreferredName())
+            .startObject(PROPERTIES)
+                .startObject(ModelPlotConfig.ENABLED_FIELD.getPreferredName())
+                    .field(TYPE, BOOLEAN)
+                .endObject()
+                .startObject(ModelPlotConfig.TERMS_FIELD.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+            .endObject()
+        .endObject()
+
+        .startObject(Job.RENORMALIZATION_WINDOW_DAYS.getPreferredName())
+            .field(TYPE, LONG) // TODO should be TimeValue
+        .endObject()
+        .startObject(Job.BACKGROUND_PERSIST_INTERVAL.getPreferredName())
+            .field(TYPE, KEYWORD)
+        .endObject()
+        .startObject(Job.MODEL_SNAPSHOT_RETENTION_DAYS.getPreferredName())
+            .field(TYPE, LONG) // TODO should be TimeValue
+        .endObject()
+        .startObject(Job.RESULTS_RETENTION_DAYS.getPreferredName())
+            .field(TYPE, LONG)  // TODO should be TimeValue
+        .endObject()
+        .startObject(Job.MODEL_SNAPSHOT_ID.getPreferredName())
+            .field(TYPE, KEYWORD)
+        .endObject()
+        .startObject(Job.RESULTS_INDEX_NAME.getPreferredName())
+            .field(TYPE, KEYWORD)
+        .endObject()
+        .startObject(Job.DELETED.getPreferredName())  // TODO can this be removed?
```

Comment:
This is an interesting question.  At the moment this is what other APIs use to determine that a job is in the process of being deleted.  Maybe storing that flag in an index won't be sufficiently atomic in the future.  An alternative might be to make the job deletion process a persistent task, and use the existence of that persistent task to determine whether a job is being deleted.  This field is probably just one of several places where indices won't give the same ordering guarantee that cluster state gave us.

Reply:
With the loose eventually consistent guarantees this field is useless, we have considered other ways to mark the fact that a job is in the process of deletion I'd like to remove this

Reply:
I've removed the `DELETED` field from the mapping